paddle/fluid/eager/backward.cc

```diff
@@ -798,15 +801,29 @@ std::vector<paddle::experimental::Tensor> RunBackward(
           bool is_potential_stop_node =
               GeneralGrad::Instance().GetPotentialStopNodes()->count(next_node);
           if (node_in_degree_map[next_node] == 0 && !is_potential_stop_node) {
-            queue.emplace(std::move(next_node));
+            if (dynamic_cast<egr::GradNodeAccumulation*>(next_node)) {
+              accumulation_node_queue.push_front(std::move(next_node));
+            } else {
+              accumulation_node_queue.push_back(std::move(next_node));
+            }
           }
         } else {
           if (node_in_degree_map[next_node] == 0) {
-            queue.emplace(std::move(next_node));
+            if (dynamic_cast<egr::GradNodeAccumulation*>(next_node)) {
+              accumulation_node_queue.push_front(std::move(next_node));
+            } else {
+              accumulation_node_queue.push_back(std::move(next_node));
+            }
           }
         }
       }
     }
+
+    while (!accumulation_node_queue.empty()) {
+      GradNodeBase* accumulation_node = accumulation_node_queue.front();
+      accumulation_node_queue.pop_front();
+      queue.push(std::move(accumulation_node));
```

Comment:
Why not use original queue to do this, avoid additional queue construction and destruction